### ISSUE TITLE
JsonConverter not being applied (on protobuf ByteString)

### ISSUE THREAD

**Issue by joshlang** [dotnet/corefx#40391]
TLDR:  My `JsonConverter<ByteString>` is never used.  `CanConvert` is never called with type `ByteString` but *is* called for `IEnumerable<byte>` instead.

...

We have a protobuf file
```
syntax = "proto3";
option csharp_namespace = "ConsoleApp1";
message Hello {
	string There = 1;
	bytes Meow = 2;
}
```

This generates a class `Hello` with two properties:  `string There` and `ByteString Meow`.  

ByteString can be found in the Google.Protobuf nuget package with definition: `public sealed class ByteString : IEnumerable<byte>, IEnumerable, IEquatable<ByteString>`

A sample object can be created like:
```
Hello hello = new Hello
{
    There = "asdf",
    Meow = ByteString.CopyFrom(1, 2, 3, 4)
};
```
…

Anyway, JsonSerializer.Serialize(hello) produces valid output.  YAY!  ❤️ System.Text.Json!

But it's not in the format I want.  So I created my converter:

```
class ByteStringConverter : JsonConverter<ByteString>
{
    public override bool CanConvert(Type typeToConvert)
    {
        bool ca = base.CanConvert(typeToConvert);
        Debug.WriteLine(typeToConvert.Name);
        return ca;
    }
    public override ByteString Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options) => throw new NotImplementedException();
    public override void Write(Utf8JsonWriter writer, ByteString value, JsonSerializerOptions options) => throw new NotImplementedException();
}
```

And used it:
```
JsonSerializerOptions o = new JsonSerializerOptions();
o.Converters.Add(new ByteStringConverter());
var s = JsonSerializer.Serialize(hello, o);
```

But it didn't work.  I expected a `NotImplementedException` but didn't get one.

CanConvert is called for 4 types:
- `Hello` (the class)
- `string` (property 'There')
- `IEnumerable<byte>` (…which is implemented by ByteString on property Meow)
- `byte`

But not for ByteString.  Which I'm assuming is why Read/Write are never called.

**Comment by joshlang** [comment]
Similarly, a protobuf field `repeated string SomeTexts = 1` generates a C# property of type `RepeatedField<string>` and it never gets passed into a CanConvert.  In this case, I used JsonConverterFactory, but that doesn't seem relevant.

Instead of `RepeatedField<string>`, it passes `ICollection<string>`.

`RepeatedField<string>` has this definition: `public sealed class RepeatedField<T> : IList<T>, ICollection<T>, IEnumerable<T>, IEnumerable, IList, ICollection, IDeepCloneable<RepeatedField<T>>, IEquatable<RepeatedField<T>>, IReadOnlyList<T>, IReadOnlyCollection<T>`

**Comment by layomia** [comment]
Thanks @joshlang!
Closing this - it's a duplicate of https://github.com/dotnet/corefx/issues/40391.
